aptos-move/aptos-vm/tests/sharded_block_executor.rs

```diff
@@ -356,9 +368,18 @@ mod test_utils {
             )
             .unwrap();
 
-        let unsharded_txn_output =
-            AptosVM::execute_block_no_limit(&execution_ordered_txns, executor.data_store())
-                .unwrap();
+        let global_cache_manager = GlobalCacheManager::new_with_default_config();
+        global_cache_manager
+            .mark_block_execution_start(executor.data_store(), None)
+            .unwrap();
+        let unsharded_txn_output = AptosVM::execute_block_no_limit(
+            &execution_ordered_txns,
+            executor.data_store(),
+            &global_cache_manager,
+        )
+        .unwrap();
+        global_cache_manager.mark_block_execution_end(None).unwrap();
```

Comment:
why not have execute_block_no_limit take Option<&GlobalCacheManager>, and if not set, create one, and then have marking done inside it?

Second this whole PR seems easier after #15137 lands - as execute_block_no_limit moves into VMBlockExecutor which has "new", and so GlobalCacheManager might live inside AptosVMBlockExecutor instance - so it doesn't even need to be passed in - if that is safe to do.

Reply:
Yes! Ideally, I want to store it in executors, but all those are stateless and we so many different ones. Let's land yours #15127 (I will do another pass), and then #15137 - I can rebase. But then we should land soonish

Reply:
@igor-aptos PRs reviewed, let's land yours so I can rebase